.github/workflows/eden_test.yml

```diff
@@ -1,10 +1,40 @@
 ---
-name: Test Eden WF from lf-edge/eden
-on:
-  workflow_dispatch:
+name: lf-edge/eden Test suite
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - "master"
+      - "[0-9]+.[0-9]+"
+      - "[0-9]+.[0-9]+-stable"
+    paths-ignore:
+      - 'docs/**'
+      - 'pkg/pillar/docs/**'
+  pull_request_review:
+    types: [submitted]
+    paths-ignore:
+      - 'docs/**'
+      - 'pkg/pillar/docs/**'
+
+concurrency:
+  group: eden_test_suite
+  cancel-in-progress: true
 
 jobs:
-  test_suite:
+  test_suite_pr:
+    if: ${{ (github.event.review.state == 'approved') &&  ( github.event_name == 'pull_request') }}
     uses: lf-edge/eden/.github/workflows/test.yml@master
     with:
       eve_image: "evebuild/danger:pr${{ github.event.pull_request.number  }}"
+
+  test_suite_master:
+    if: ${{ github.ref == 'refs/heads/master' }}
+    uses: lf-edge/eden/.github/workflows/test.yml@master
+    with:
+      eve_image: "lfedge/eve:snapshot"
+
+  test_suite_tag:
+    if: ${{ startsWith(github.ref, 'refs/tags') }}
+    uses: lf-edge/eden/.github/workflows/test.yml@master
```

Comment:
Is it possible to use the test.yml from the branch/tag? We might have new tests in master which would not work on the older release branch.

Reply:
Yes, that's the plan. @uncleDecart will create a tag for the eden tests so that master can be used for modifications

Reply:
@eriknordmark I started [conversation](https://lists.lfedge.org/g/eve/topic/eden_eve_api_eve_libs/100936104?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,100936104,previd%3D1694669988883123554,nextid%3D1685561082904173507&previd=1694669988883123554&nextid=1685561082904173507) on eve email list, if there are no objections from your side, I'll release eden version 0.9.0 and will submit PR to reference 0.9.0 tag for this workflow

Reply:
@uncleDecart Please let me know once the tag is released, I will modify the reference branch in eve WF.